laurent22/joplin
# Mobile: Fixes #10143: Shows only the real folders in the dropdown of parent folders.

## Summary

On creating a new notebook/note, the dropdown of list of parent folders shows only the real folders, the trash folder and the deleted folders are filtered out.

- Fixes #10143 

## Rationale

Created a function `getRealFolders` which takes an array of folders as an input and returns it after filtering the trash folder and deleted folders. Used this function on the folders array before passing it to the `FolderPicker` component.

## Demo


https://github.com/laurent22/joplin/assets/97161173/123c6b8e-11f2-4448-9b0a-562310d79ba6




## Testing
Tested on Android 13 physical device.
1. Create 2 notebooks and delete one of it.
2. Try to create another notebook and check the dropdown for selecting parent folder, the deleted folder should not appear.
3. Repeat step 2 for notes and todos.


@PackElend label me please!
